packages/react-mutation-mapper/src/store/DefaultMutationMapperStore.ts

```diff
@@ -649,10 +659,66 @@ class DefaultMutationMapperStore implements MutationMapperStore {
     );
 
     readonly ptmData: MobxPromise<PostTranslationalModification[]> = remoteData(
+        {
+            await: () => [this.uniprotPtmData],
+            invoke: () => {
+                let data: PostTranslationalModification[] = [];
+
+                if (
+                    this.ptmSource === PtmSource.Uniprot &&
+                    this.uniprotPtmData.result
+                ) {
+                    data = this.uniprotPtmData.result.map(
+                        convertUniprotFeatureToPtm
+                    );
+                } else if (
+                    this.ptmSource === PtmSource.dbPTM &&
+                    this.dbPtmData.result
+                ) {
+                    data = this.dbPtmData.result.map(convertDbPtmToPtm);
+                }
+
+                return Promise.resolve(data);
+            },
+            onError: () => {
+                // fail silently
+            },
+        }
+    );
+
+    readonly uniprotPtmData: MobxPromise<UniprotFeature[]> = remoteData(
+        {
+            await: () => [
+                this.mutationData,
+                this.activeTranscript,
+                this.swissProtId,
+            ],
+            invoke: async () => {
+                if (this.activeTranscript.result && this.swissProtId.result) {
+                    // TODO we need to update genome nexus for this one to work,
+                    //  for now we are getting the data directly from uniprot API
+                    // return this.dataFetcher.fetchPtmData(
+                    //     this.activeTranscript.result
+                    // );
+                    return this.dataFetcher.fetchUniprotFeatures(
+                        this.swissProtId.result
+                    );
+                } else {
+                    return [];
+                }
+            },
+            onError: () => {
+                // fail silently
+            },
+        },
+        []
+    );
+
+    readonly dbPtmData: MobxPromise<DbPtm[]> = remoteData(
         {
             await: () => [this.mutationData, this.activeTranscript],
             invoke: async () => {
-                if (this.activeTranscript.result) {
+                if (this.activeTranscript.result && this.swissProtId.result) {
```

Comment:
Why do we need to check `swissProtId` here?

Reply:
swissProtId is needed to get the uniprot accession. But this is temporary, we should be only using the transcript id when querying for the PTMs.